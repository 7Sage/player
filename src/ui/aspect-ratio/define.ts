import { safelyDefineCustomElement } from '../../utils/dom';
import {
  ASPECT_RATIO_ELEMENT_TAG_NAME,
  AspectRatioElement
} from './AspectRatioElement';

safelyDefineCustomElement(ASPECT_RATIO_ELEMENT_TAG_NAME, AspectRatioElement);

declare global {
  interface HTMLElementTagNameMap {
    [ASPECT_RATIO_ELEMENT_TAG_NAME]: AspectRatioElement;
  }
}
