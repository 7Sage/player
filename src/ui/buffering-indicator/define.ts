import { safelyDefineCustomElement } from '@utils/dom';

import {
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
} from './BufferingIndicatorElement';

safelyDefineCustomElement(
  BUFFERING_INDICATOR_ELEMENT_TAG_NAME,
  BufferingIndicatorElement
);

declare global {
  interface HTMLElementTagNameMap {
    [BUFFERING_INDICATOR_ELEMENT_TAG_NAME]: BufferingIndicatorElement;
  }
}
