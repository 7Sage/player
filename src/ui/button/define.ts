import { safelyDefineCustomElement } from '../../utils/dom';
import { BUTTON_ELEMENT_TAG_NAME, ButtonElement } from './ButtonElement';

safelyDefineCustomElement(BUTTON_ELEMENT_TAG_NAME, ButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    [BUTTON_ELEMENT_TAG_NAME]: ButtonElement;
  }
}
