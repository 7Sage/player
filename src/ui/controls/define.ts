import { safelyDefineCustomElement } from '@utils/dom';

import { CONTROLS_ELEMENT_TAG_NAME, ControlsElement } from './ControlsElement';

safelyDefineCustomElement(CONTROLS_ELEMENT_TAG_NAME, ControlsElement);

declare global {
  interface HTMLElementTagNameMap {
    [CONTROLS_ELEMENT_TAG_NAME]: ControlsElement;
  }
}
