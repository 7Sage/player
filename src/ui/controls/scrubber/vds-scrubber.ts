// ** Dependencies **
import '../slider/vds-slider';

import { LIB_PREFIX } from '../../../shared/constants';
import { safelyDefineCustomElement } from '../../../utils/dom';
import { Scrubber } from './Scrubber';

export const SCRUBBER_TAG_NAME = `${LIB_PREFIX}-scrubber` as const;

safelyDefineCustomElement(SCRUBBER_TAG_NAME, Scrubber);

declare global {
  interface HTMLElementTagNameMap {
    [SCRUBBER_TAG_NAME]: Scrubber;
  }
}
