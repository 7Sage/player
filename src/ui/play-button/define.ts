import { safelyDefineCustomElement } from '@utils/dom';

import {
  PLAY_BUTTON_ELEMENT_TAG_NAME,
  PlayButtonElement
} from './PlayButtonElement';

safelyDefineCustomElement(PLAY_BUTTON_ELEMENT_TAG_NAME, PlayButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    [PLAY_BUTTON_ELEMENT_TAG_NAME]: PlayButtonElement;
  }
}
