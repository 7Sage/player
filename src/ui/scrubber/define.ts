import { safelyDefineCustomElement } from '../../utils/dom';
import { SCRUBBER_ELEMENT_TAG_NAME, ScrubberElement } from './ScrubberElement';

safelyDefineCustomElement(SCRUBBER_ELEMENT_TAG_NAME, ScrubberElement);

declare global {
  interface HTMLElementTagNameMap {
    [SCRUBBER_ELEMENT_TAG_NAME]: ScrubberElement;
  }
}
