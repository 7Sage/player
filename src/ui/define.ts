import './buffering-indicator/define';
import './controls/define';
import './fullscreen-button/define';
import './mute-button/define';
import './play-button/define';
import './scrim/define';
import './scrubber/define';
import './scrubber-preview/define';
import './scrubber-preview-time/define';
import './scrubber-preview-video/define';
import './seekable-progress-bar/define';
import './slider/define';
import './time/define';
import './time-current/define';
import './time-duration/define';
import './time-slider/define';
import './toggle-button/define';
import './volume-slider/define';

export * from '.';
