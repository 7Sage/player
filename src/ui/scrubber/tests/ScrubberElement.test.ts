import '../../../define/vds-scrubber';

import { getDiffableSemanticHTML } from '@open-wc/semantic-dom-diff';
import { html } from 'lit';

import { buildMediaPlayerFixture } from '../../../media/test-utils';

async function buildFixture() {
  const { player } = await buildMediaPlayerFixture(html`
    <vds-scrubber></vds-scrubber>
  `);

  const scrubber = player.querySelector('vds-scrubber')!;

  return { scrubber };
}

test('light DOM snapshot', async function () {
  const { scrubber } = await buildFixture();
  expect(scrubber).dom.toMatchSnapshot();
});

test('shadow DOM snapshot', async function () {
  const { scrubber } = await buildFixture();
  expect(
    getDiffableSemanticHTML(scrubber.shadowRoot!.firstElementChild!)
  ).toMatchSnapshot();
});
