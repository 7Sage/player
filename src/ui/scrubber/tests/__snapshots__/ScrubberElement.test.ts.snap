// Vitest Snapshot v1

exports[`light DOM snapshot 1`] = `
<vds-scrubber
  aria-label="Media time"
  aria-orientation="horizontal"
  aria-valuemax="0"
  aria-valuemin="0"
  aria-valuenow="0"
  aria-valuetext="0 seconds out of 0 seconds"
  autocomplete="off"
  role="slider"
  tabindex="0"
/>
`;

exports[`shadow DOM snapshot 1`] = `
"<div
  id=\\"root\\"
  part=\\"root\\"
  role=\\"presentation\\"
  style=\\"--vds-slider-fill-value: 0; --vds-slider-fill-rate: 0; --vds-slider-fill-percent: 0%; --vds-slider-drag-rate: 0; --vds-slider-drag-percent: 0%;\\"
>
  <div
    id=\\"thumb-container\\"
    part=\\"thumb-container\\"
  >
    <div
      id=\\"thumb\\"
      part=\\"thumb\\"
    >
      <slot name=\\"thumb\\">
      </slot>
    </div>
    <slot name=\\"thumb-container\\">
    </slot>
  </div>
  <div
    id=\\"track\\"
    part=\\"track\\"
  >
    <slot name=\\"track\\">
    </slot>
  </div>
  <div
    id=\\"track-fill\\"
    part=\\"track-fill\\"
  >
    <slot name=\\"track-fill\\">
    </slot>
  </div>
  <input
    max=\\"100\\"
    min=\\"0\\"
    type=\\"hidden\\"
    value=\\"0\\"
  >
  <slot>
  </slot>
</div>
"
`;
